packages/trpc/server/routers/publicViewer/stripeCheckoutSession.handler.ts

```diff
@@ -10,12 +11,9 @@ export const stripeCheckoutSessionHandler = async ({ input }: StripeCheckoutSess
   const { checkoutSessionId, stripeCustomerId } = input;
 
   // TODO: Move the following data checks to superRefine
-  if (!checkoutSessionId && !stripeCustomerId) {
-    throw new Error("Missing checkoutSessionId or stripeCustomerId");
-  }
-
-  if (checkoutSessionId && stripeCustomerId) {
-    throw new Error("Both checkoutSessionId and stripeCustomerId provided");
+  const validationResult = ZStripeCheckoutSessionInputSchema.safeParse(input);
+  if (!validationResult.success) {
+    throw new Error(validationResult.error.message);
```

Comment:
Looks good to me, but why not use .parse if you are throwing the error anyway? Unhappy with the default zod error?